trieste/models/interfaces.py

```diff
@@ -226,3 +249,91 @@ def optimize(self, dataset: Dataset) -> None:
 
         for model, obs in zip(self._models, observations):
             model.optimize(Dataset(dataset.query_points, obs))
+
+
+
+    def reparam_sampler(self, num_samples: int) -> ReparametrizationSampler:
+        """
+        Return a reparametrization sampler providing `num_samples` samples.
+
+        Note that this is not supported by all models.
+
+        :param num_samples: The desired number of samples.
+        :return: The reparametrization sampler.
+        """
+        raise NotImplementedError(f"Model {self!r} does not have a reparametrization sampler")
+
+
```

Comment:
I feel like it would be better to have the base classes for the samplers in their own file `samplers.py`, but this would probably cause circular imports?

Reply:
I agree, samplers are distinct enough to warrant a separate file - what about only a generic `ModelSampler` class defined here and used above as output type, and move all of these to a `sampler.py` file (as subclasses) - that would avoid circular import issue